nixos/modules/services/web-apps/keycloak.nix

```diff
@@ -0,0 +1,71 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.keycloak;
+in {
+  options = {
+    services.keycloak = {
+      enable = mkEnableOption "Keycloak identity and access server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.keycloak;
+        defaultText = "pkgs.keycloak";
+        example = literalExample "pkgs.keycloak";
+        description = "Package that should be used for keycloak";
+      };
+
+      jbossBaseDir = mkOption {
+        type = types.str;
+        default = "/var/lib/keycloak";
+        description = ''
+          The location keycloak (and underlying JBoss/Wildfly) information
+          (configuration, logs, data) is stored.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        description = "The user to run keycloak as.";
+        default = "keycloak";
+      };
+
+      group = mkOption {
+        type = types.str;
+        description = "The group to run keycloak as.";
+        default = "keycloak";
+      };
+
+      extraFlags = mkOption {
+        description = "Extra flags to pass to the keycloak (standalone.sh) command.";
+        default = "";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.keycloak-server = {
```

Comment:
You'll want to add reference to `multi-user.target` in here as well (`wantedBy`).

Reply:
fixed